### PR TITLE
Implement HPE redfish firmware push

### PR DESCRIPTION
**Pull request by Arno500**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

It should fix and implement the required procedures to push a firmware on iLO 5 and 6 (tested on iLO 5 only for now).
Supersede #8478

### CODE REVIEW

**Comment by hughsie** [comment]
@Arno500 I've fixed up the abstraction a bit and tidied the commit descriptions a little -- I hope that's okay. Can you check this still works on real hardware please, and then after that LGTM. Good job!

**Review by hughsie — APPROVED** [review]

**Comment by Arno500** [comment]
Thanks @hughsie, I will check that on monday!
(and nope, tidying my work is more than welcome!)

File: plugins/redfish/fu-self-test.c [hunk 1]
```diff
@@ -87,6 +92,35 @@ fu_test_self_init(FuTest *self)
...
+	if (g_error_matches(error, FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE)) {
+		g_test_skip("no redfish.py running");
+	} else {
+		g_assert_no_error(error);
+		g_assert_true(ret);
+
+		fu_redfish_plugin_set_credentials(self->hpe_plugin, "hpe_username", "password2");
+		/* We just changed the credentials and need to resetup again to discover the vendor
+		 * Here we need to get a device to query the backend
+		 */
```
Comment (superm1) [inline plugins/redfish/fu-self-test.c:108]: It doesn't need to be for this PR; but for the purpose of test coverage I think it would be a good idea to also have a negative test (IE send a bad user and then send a bad password)

**Review by superm1 — APPROVED** [review]

File: plugins/redfish/fu-self-test.c (hunk 1 above)
Comment (Arno500) [reply plugins/redfish/fu-self-test.c:108]: That would probably be a good idea "redfish wise", yes

**Comment by Arno500** [comment]
Can confirm everything works as expected, I will let you click on the ~red~ green button
